tests/it/array/growable/dictionary.rs

```diff
@@ -29,6 +30,24 @@ fn test_single() -> Result<()> {
     Ok(())
 }
 
+#[test]
+fn test_negative_keys() {
+    let vals = vec![Some("a"), Some("b"), Some("c")];
+    let keys = vec![0, 1, 2, -1];
+
+    let keys = PrimitiveArray::from_data(
+        DataType::Int32,
+        keys.into(),
+        Some(vec![true, true, true, false].into()),
+    );
+
+    let arr = DictionaryArray::from_data(keys, Arc::new(Utf8Array::<i32>::from(vals)));
+    // check that we don't panic with negative keys to usize conversion
+    let out = arrow2::compute::concat::concatenate(&[&arr]).unwrap();
```

Comment:
Can we replace this call by the growable API?

`cargo test --tests` (without features) is very useful to run when not iterating over the compute module kernels, and this usage causes it to break.